### PR TITLE
Add questions vs answers status bar on tag stats page

### LINKED ISSUE #6790: Add questions asked vs asked status bar in tag/topic page
Sub task of https://github.com/publiclab/plots2/issues/2120.

### Description
---------------------
Add a questions asked vs answered rate  on https://publiclab.org/tag/community/stats. We should reuse the dat picker there to calculate the rate while calculating the other existing metrics.

We already have a silmilar status bar here Screenshot from 2019-11-17 12-22-22  on https://publiclab.org/stats?options=Past+year and you can reuse the idea used there. 

### Where to find Logic and templates 
--------------------------------------------------------
We get the questions vs answers with like this https://github.com/publiclab/plots2/blob/master/app/controllers/stats_controller.rb#L28-L34

The graph code can be found here... https://github.com/publiclab/plots2/blob/master/app/views/stats/index.html.erb#L259%20-L326 .
You could extract this as a partial and reuse for the two views.

The view that we need to add the status bar is https://github.com/publiclab/plots2/blob/master/app/views/tag/stats.html.erb and its action in controller is https://github.com/publiclab/plots2/blob/master/app/controllers/tag_controller.rb#L524-L535

### PR DESCRIPTION
**Pull request by nstjean**
Fixes #6790  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by nstjean** [comment]
I added `Node.find_by_tag(tagname)` to reduce code duplication, it simply returns all nodes with that tagname. I looked for an existing function, there is `filter_questions_by_tag` in Questions Controller that does a similar thing, but it's a private function so can't be used elsewhere.

I tried to clean things up a little bit. I also added a default width of 100% for the status bar, but allowed it to be overridden by adding `data-width="91%"` to the div, since in stats/index it was set at 91% to match the rest of the bars on the page.

**Comment by nstjean** [comment]
Here's a screenshot of localhost:3000/tag/trees/stats

FireShot Capture 065 - 🎈 Public Lab_ Stats - localhost

**Comment by nstjean** [comment]
Looking at the page I just realized that the numbers as it was on the original page are wrong. it's showing 2 questions and 1 answers, so the bar should be showing 1/2 complete. but it's showing 1/3 complete. I'll have to look more at how it's calculating things!

**Comment by nstjean** [comment]
Ok so I made changes! The old function was using the depreciated Answer model, so I've changed it to counting questions that have comments.

The status bar now shows a total of 2 questions, 1 is unanswered, 1 has answers (comments). Does that sound right? Maybe we should put the answered questions number on the left side, so the bar will progress left to right until all questions are answered?

FireShot Capture 066 - 🎈 Public Lab_ Stats - localhost

**Comment by nstjean** [comment]
@cesswairimu What do you think? Should we have the answered number on the left side or just leave it as it is?

**Comment by cesswairimu** [comment]
Not sure why travis is getting the sprockets error..restarting server

**Comment by nstjean** [comment]
Ok I switched the bars, edited the titles to match, and removed that weird "1" that was the title of the tooltip that pops up on hover.

FireShot Capture 068 - 🎈 Public Lab_ Stats - localhost

FireShot Capture 069 - 🎈 Public Lab_ Stats - localhost


On hover:
FireShot Capture 067 - 🎈 Public Lab_ Stats - localhost

**Comment by jywarren** [comment]
Beautiful! Is this ready to merge then? Thank you!

If you'd like, you could add an example tag page to our system test
screenshots - in /test/system/screenshots_test.rb - there are other
examples there that should be easy to copy. Then Travis will generate
screenshots of this page for each PR!

**Comment by nstjean** [comment]
Sure! Should I add that screenshot test in a separate PR?

**Comment by jywarren** [comment]
You can! Or you can add it here. Either works!

**Comment by cesswairimu** [comment]
Sweet! @nstjean could we change the "Unanswered" to Asked because we are getting the tally of the questions asked within a certain period..(some of which could be also answered). Makes sense?

**Comment by nstjean** [comment]
@cesswairimu Ahhh, do you want it comparing to the total number of questions then? I had changed it to calculate the number of unanswered questions instead, so that the bar could be totally full (all questions answered). If it stays answered vs total number it will show as halfway if all the questions are answered. So that's up to you what you want!

File: app/controllers/stats_controller.rb [hunk 1]
```diff
@@ -25,15 +25,16 @@ def range
...
-        .size
       @comments = Comment.select(:status, :timestamp)
         .where(status: 1, timestamp: @start.to_i..@end.to_i)
         .size
-      @questions = Node.published.questions.where(created: @start.to_i..@end.to_i)
-        .size
       @contributors = User.contributor_count_for(@start, @end)
       @popular_tags = Tag.nodes_frequency(@start, @end)
+
+      total_questions = Node.published.questions
```
Comment (cesswairimu) [inline app/controllers/stats_controller.rb]: Am sorrry @nstjean, I believe this is getting all the questions published?

File: app/views/stats/_barchart.html.erb [hunk 2]
```diff
@@ -0,0 +1,77 @@
+
+<script src="https://cdn.jsdelivr.net/npm/apexcharts"></script>
+<script>
+var questions = [<%= @questions.present? ? @questions : 0 %>];
+var answers = [<%= @answers %>];
+var unanswered = [ questions[0] - answers[0] ];
```
Comment (cesswairimu) [inline app/views/stats/_barchart.html.erb]: Oh I see you do the subtraction here...my bad

File: app/views/stats/_barchart.html.erb [hunk 3]
```diff
@@ -0,0 +1,77 @@
+
+<script src="https://cdn.jsdelivr.net/npm/apexcharts"></script>
+<script>
+var questions = [<%= @questions.present? ? @questions : 0 %>];
+var answers = [<%= @answers %>];
```
Comment (nstjean) [inline app/views/stats/_barchart.html.erb]: Yes `@questions` does get all of them. But here in barchart I am calculating the unanswered. It's easy to change back!

File: app/views/stats/_barchart.html.erb (hunk 2 above)
Comment (nstjean) [reply app/views/stats/_barchart.html.erb]: Yep there it is! :)

File: app/views/stats/_barchart.html.erb (hunk 3 above)
Comment (cesswairimu) [reply app/views/stats/_barchart.html.erb]: Great, this is awesome, thanks so much so working on this :tada:

**Comment by jywarren** [comment]
Awesome work all!!!!

**Pull request merged**